pkg/client-sdk/client.go

```diff
@@ -142,6 +143,35 @@ func (a *arkClient) ListVtxos(
 	return
 }
 
+func (a *arkClient) NotifyIncomingFunds(
+	ctx context.Context, addr string,
+) ([]types.Vtxo, error) {
+	eventCh, closeFn, err := a.client.SubscribeForAddress(ctx, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
```

Comment:
This `wg` is weird:
```
wg.Add(1)
go func() ... 
wg.Done()
```

it seems it is never waited and done is called just after starting the routine, sounds we don't need it